cmd/argocd/commands/cluster.go

```diff
@@ -238,17 +241,21 @@ func NewClusterSetCommand(clientOpts *argocdclient.ClientOptions) *cobra.Command
 			conn, clusterIf := headless.NewClientOrDie(clientOpts, c).NewClusterClientOrDie()
 			defer io.Close(conn)
 			// checks the fields that needs to be updated
-			updatedFields := checkFieldsToUpdate(clusterOptions)
+			updatedFields := checkFieldsToUpdate(clusterOptions,labels)
 			namespaces := clusterOptions.Namespaces
 			// check if all namespaces have to be considered
 			if len(namespaces) == 1 && strings.EqualFold(namespaces[0], allNamespaces) {
 				namespaces[0] = ""
 			}
+			// parse the labels you're receiving from the label flag
+			labelsMap, err := label.Parse(labels)
+			errors.CheckError(err)
 			if updatedFields != nil {
 				clusterUpdateRequest := clusterpkg.ClusterUpdateRequest{
 					Cluster: &argoappv1.Cluster{
 						Name:       clusterOptions.Name,
 						Namespaces: namespaces,
+						Labels:		labelsMap,
```

Comment:
```suggestion
						Labels:		labelsMap,
```

Inspite of updating the labels, the labels do not get updated on argocd even though the update response is 200.

Reply:
As i understand, you mean that you passing value, but server is not updating it?

Reply:
yes, it is not updating it in spite of passing the labels.

Reply:
The interesting thing to note here is that if i try to update any other field, say name, it gets updated successfully. However, the lables are not getting updated inpsite of passing them via CLI.